code/modules/reagents/chemistry/holder/mob_life.dm

```diff
@@ -16,16 +21,22 @@
 	var/need_mob_update = FALSE
 	var/obj/item/organ/internal/stomach/belly = owner.get_organ_slot(ORGAN_SLOT_STOMACH)
 	var/obj/item/organ/internal/liver/liver = owner.get_organ_slot(ORGAN_SLOT_LIVER)
-	var/liver_tolerance
+	var/liver_tolerance = 0
+	var/liver_damage = 0
+	var/provide_pain_message
+	var/amount
 	if(liver)
 		var/liver_health_percent = (liver.maxHealth - liver.damage) / liver.maxHealth
 		liver_tolerance = liver.toxTolerance * liver_health_percent
+		provide_pain_message = HAS_NO_TOXIN
 
 	for(var/datum/reagent/reagent as anything in cached_reagents)
```

Comment:
```suggestion
	for(var/datum/reagent/toxin/toxin in cached_reagents)
```

If you need type checking don't use `as anything`

Reply:
Whatever speed you get from as anything is negated by the istype() and null checks for the toxin var next line.

In fact I think it's slower now because of it

Reply:
> the code you gave me quite literally breaks all of reagents metabolizing. 

You were meant to make the change and then see all effected code, not just stop with my suggestion is what I meant

Reply:
Ok so I made an incomplete suggestion which caused compile errors, I'll make the full suggestion this time to clear up all confusion

Reply:
You right, Sorry I can't read today.

Carry on with your day